elasticsearch/client.go

```diff
@@ -1,26 +1,81 @@
 package elasticsearch
 
-// Client provides methods to wrap around the elastcsearch package in order to facilitate unit testing
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	rchttp "github.com/ONSdigital/dp-rchttp"
+)
+
+// Client represents an instance of the elasticsearch client
 type Client struct {
-	URL string
+	url          string
+	rchttpClient rchttp.Clienter
 }
 
 // New creates a new elasticsearch client
-func New(URL string) *Client {
-
-	Setup(URL)
-
+func New(url string, rchttpClient rchttp.Clienter) *Client {
 	return &Client{
-		URL: URL,
+		url:          strings.TrimRight(url, "/"),
```

Comment:
I would include a note about the url trim slash in the godoc for this method. Very minor point but then the godoc for this method is being 100% transparent about what it does so the caller knows exactly what to expect..